models/issue.go

```diff
@@ -1790,8 +1798,8 @@ func GetRepoIssueStats(repoID, uid int64, filterMode int, isPull bool) (numOpen
 }
 
 // SearchIssueIDsByKeyword search issues on database
-func SearchIssueIDsByKeyword(kw string, repoID int64, limit, start int) (int64, []int64, error) {
-	var repoCond = builder.Eq{"repo_id": repoID}
+func SearchIssueIDsByKeyword(kw string, repoIDs []int64, limit, start int) (int64, []int64, error) {
+	var repoCond = builder.In("repo_id", repoIDs)
 	var subQuery = builder.Select("id").From("issue").Where(repoCond)
```

Comment:
Perhaps you should just search in the first repo (usually the local repo) and only if limit is not reached, then search in the other repos. In either case you should at least sort locally by ID descending:
```
import "sort"
sort.Slice(ids, func(i, j int) bool {return ids[i] > ids[j]})
```

Reply:
this makes sense when using the db option for issue indexing, but if using bleve, I'm guessing I should leave that as is, since by default my understanding is that the sort is by relevancy scoring, but I could be corrected. And I think this code is only hit when a keyword is actually typed, otherwise it is just a list of issues up to the limit (i.e. not running this query) so I'll visit that code too and see what the best way to get a predictable listing is.

Reply:
I see. I guess there's not much to do about the order returned by Bleve. However, it is possible to prioritize the local repository because Bleve keeps separate indexes for each one and the search is done one by one. I didn't check, perhaps it comes out naturally?

Reply:
I looked at it a little more tonight and I think I have a better plan to make it straightforward I hope. I feel a little silly but I believe the ordering is actually coming out of a call to models.Issues where we pass the issue ids after we've found the list, so the sorting in these search methods isn't important. I think I can apply the proper sorting there so it would be consistent for db or bleve. Hopefully in a few days I'll be able to commit some changes that hopefully resolve the issues you pointed out related to this thanks!

Reply:
I believe I fixed the sorting of the issues the way you proposed but it isn't done here.